aws/resource_aws_networkfirewall_rule_group.go

```diff
@@ -511,7 +511,7 @@ func resourceAwsNetworkFirewallRuleGroupUpdate(ctx context.Context, d *schema.Re
 		if v, ok := d.GetOk("description"); ok {
 			input.Description = aws.String(v.(string))
 		}
-		if v, ok := d.GetOk("rule_group"); ok {
+		if v, ok := d.GetOk("rule_group"); ok && !d.HasChange("rules") {
```

Comment:
so i though i would be able to do something like 
```suggestion
		if d.HasChange("rule_group") {
		  ...
		  input.RuleGroup = ...
	        }
	        
	        if d.HasChange("rules") {
	         ...
	         input.Rules = ...
	        }
```
but the Computed nature of `rule_group` returns true for `d.HasChange()`